src/switch_msrp.c

```diff
@@ -252,15 +252,26 @@ static switch_status_t msock_init(char *ip, switch_port_t port, switch_socket_t
 	switch_sockaddr_t *sa;
 	switch_status_t rv;
 
-	rv = switch_sockaddr_info_get(&sa, ip, SWITCH_INET, port, 0, pool);
-	if (rv) goto sock_fail;
+	rv = switch_sockaddr_info_get(&sa, ip, SWITCH_UNSPEC, port, 0, pool);
+	if (rv) {
+		switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_ERROR, "Cannot get information about MSRP listen IP address %s\n", ip);
+		goto sock_fail;
+	}
 
 	rv = switch_socket_create(sock, switch_sockaddr_get_family(sa), SOCK_STREAM, SWITCH_PROTO_TCP, pool);
 	if (rv) goto sock_fail;
 
 	rv = switch_socket_opt_set(*sock, SWITCH_SO_REUSEADDR, 1);
 	if (rv) goto sock_fail;
 
+#ifdef WIN32
+	/* Enable dual-stack listening on Windows */
+	if (switch_sockaddr_get_family(sa) == AF_INET6) {
+		rv = switch_socket_opt_set(*sock, 16384, 0);
```

Comment:
Please add this definition to the `switch_apr.h`
```
#define SWITCH_SO_IPV6_V6ONLY 16384 /* Don't accept IPv4 connections on an IPv6 listening socket. */
```

Reply:
Appreciate the advice! I took the liberty to update the other IPv6 use cases so the code is more readable/consistent.